examples/fleet.yaml

```diff
@@ -95,7 +100,7 @@ spec:
       # Now in Beta, and enabled by default
       counters:
         players:
```

Comment:
Ow, I just realised my head hurts. I thought I'd fixed this everywhere when writing https://agones.dev/site/docs/guides/counters-and-lists/ , but clearly I missed some.

What we should have in all examples and docs:

`rooms` - is a counter
`players` is a list.

Somewhere this all got messed up, with an `rooms` being a list, and the introduction of `sessions` - but the above landing page, that should be the source of truth.

Sorry to make you go back and forth - I just noticed this.

Let's stick to `players` and `rooms` and remove all else please 😄

Reply:
Hey no worries at all! Anything to make to make the docs better 🕺.

I have now removed mentions of `sessions` in and made `rooms` only used as an example as counters files e.g., `fleet.yaml` and `gameserver.yaml` in examples and `fleet.md` and `gameserver.md` in `Reference`. 

However as a result of this examples for lists using rooms which showed initial values were also removed. If the intention of this was to show user's they are able to set initial values, I can revert this and change the wording from `rooms` to maybe `map` and initial values to be `map1`, `map2`, etc.

<img width="461" alt="Screenshot 2024-05-18 at 17 57 45" src="https://github.com/googleforgames/agones/assets/38630464/22da0f43-69c4-4c51-b76f-a24637165d53">

Reply:
I'd prefer to leave in some mention of the above either in the example or the docs. The idea is to show that you can the same key in Counters as in List, that a List can have starting values, and that certain values may be nil.

Reply:
Okie dokes. As that is the intention, I have reintroduced them in `gameserver.yaml` in `examples` and `gameserver.md` in `References` since GameServer would be the source of most user's reference when viewing the docs in addition to it being linked from the `CountersAndLists` documentation page itself.

Commit: 07e491db9e0b2b2543ecfd6498f0e14dd8c35b4e